sdk/communication/azure-communication-phonenumbers/azure/communication/phonenumbers/siprouting/_sip_routing_client.py

```diff
@@ -0,0 +1,259 @@
+
+from._models import SipTrunk
+from ._generated.models import (
+    SipConfiguration,
+    SipTrunkRoute,
+    SipTrunkInternal
+)
+from ._generated._sip_routing_service import SIPRoutingService
+from .._shared.utils import (
+    parse_connection_str,
+    get_authentication_policy
+)
+from .._version import SDK_MONIKER
+
+if TYPE_CHECKING:
+    from typing import Iterable, List, Any
+    from azure.core.credentials import TokenCredential
+
+
+class SipRoutingClient(object):
+    """A client to interact with the AzureCommunicationService SIP routing gateway.
+    This client provides operations to retrieve and manage SIP routing configuration.
+
+    :param endpoint: The endpoint url for Azure Communication Service resource.
+    :type endpoint: str
+    :param credential: The credentials with which to authenticate.
+    :type credential: TokenCredential
+    :keyword api_version: Api Version. Default value is "2021-05-01-preview". Note that overriding
+     this default value may result in unsupported behavior.
+    :paramtype api_version: str
+    """
+
+    def __init__(
+        self,
+        endpoint,  # type: str
+        credential,  # type: TokenCredential
+        **kwargs  # type: Any
+    ):  # type: (...) -> SipRoutingClient
+
+        if not credential:
+            raise ValueError("credential can not be None")
+        try:
+            if not endpoint.lower().startswith("http"):
+                endpoint = "https://" + endpoint
+        except AttributeError:
+            raise ValueError("Host URL must be a string")
+
+        parsed_url = urlparse(endpoint.rstrip("/"))
+        if not parsed_url.netloc:
+            raise ValueError("Invalid URL: {}".format(endpoint))
+
+        self._endpoint = endpoint
+        self._authentication_policy = get_authentication_policy(endpoint, credential)
+
+        self._rest_service = SIPRoutingService(
+            self._endpoint,
+            authentication_policy=self._authentication_policy,
+            sdk_moniker=SDK_MONIKER,
+            **kwargs
+        )
+
+    @classmethod
+    def from_connection_string(
+        cls,
+        conn_str,  # type: str
+        **kwargs  # type: Any
+    ):
+        # type: (...) -> SipRoutingClient
+        """Factory method for creating client from connection string.
+
+        :param str conn_str: Connection string containing endpoint and credentials.
+        :returns: The newly created client.
+        :rtype: ~azure.communication.siprouting.SipRoutingClient
+        """
+        endpoint, credential = parse_connection_str(conn_str)
+        return cls(endpoint, credential, **kwargs)
+
+    @distributed_trace
+    def get_trunk(
+        self,
+        trunk_fqdn,  # type: str
+        **kwargs  # type: Any
+    ):  # type: (...) -> SipTrunk
+        """Getter for single SIP trunk.
+
+        :param trunk_fqdn: FQDN of the desired SIP trunk.
+        :type trunk_fqdn: str
+        :returns: SIP trunk with specified trunk_fqdn.
```

Comment:
We should elaborate that if the specified trunk is not found, None is returned.

Reply:
Fixed.